src/chai/assertion.ts

```diff
@@ -0,0 +1,226 @@
+/*!
+ * chai
+ * http://chaijs.com
+ * Copyright(c) 2011-2014 Jake Luer <jake@alogicalparadox.com>
+ * MIT Licensed
+ */
+
+import {config} from './config.js';
+import {AssertionError} from 'assertion-error';
+import * as util from './utils/index.js';
+import {ChainableBehavior} from './utils/chainableBehavior.js';
+
+export interface AssertionFlags {
+  ssfi: Function;
+  lockSsfi?: boolean;
+  object: unknown;
+  message?: string | null;
+  eql: (a: unknown, b: unknown) => boolean;
+  [key: PropertyKey]: unknown;
+}
+
+type MethodNames<T> = {
+  [k in keyof T]: T[k] extends (...args: never) => void ? k : never;
+}[keyof T];
+
+const getDefaultValue = (assertion: Assertion): Assertion => {
```

Comment:
without this, we end up with circular imports: `assertion.js` -> `addMethod.js` -> `assertion.js`.

it is probably a bad idea to have such a circular dependency, so i extracted it into a factory function we now have to pass in (if we want an assertion to be the return value).

if anyone has a better idea, i'm all ears